### ISSUE TITLE
App purchased from microsoft store does not show as activated on startup 

### ISSUE THREAD

**Issue by benrobbins20** [WindowTop/WindowTop-App#290]
Submitted customer support request but has not been resolved, uninstalled and install setup from gitub releases. Did not resolve issue. How would someone access their product key when purchased from microsoft store?

**Comment by gileli121** [comment]
@benrobbins20 
Thank you for contacting us again.
There was a misunderstanding here.
I thought that you purchased an activation key on the website.

What you did was purchase it via Microsoft Store directly.
When purchasing via Microsoft Store, the activation method works differently:
Instead of serial-key-based, it is based on your Microsoft account, so in this method, you don't need a key to activate the app.
Instead, you need to login into the Microsoft store via WindowTop.

Unfortunately, I did not add a "Login to MS" in the app. it is a design flaw.
So a user who purchased the app and, at some point, reinstalled Windows or something like that, and then installed the app via Microsoft Store - may be confused about why the app is not activated yet. and this is because the app does not tell and give the user a simple, straightforward way to login to his account (where he owns it)

Anyway, I will work on implementing the login button (or whatever it should be called) so users will not be confused like this again.


As for now, you can still trigger the login flow by trying to purchase the app again:
1. Click on the "Purchase (Microsoft Store)" button
2. Once you click the button, it will trigger the purchase flow, including the login. It will ask you to provide your username and password for the Microsoft store
3. After you log in, you must purchase it again. Don't worry; Microsoft Store services will not let you make a second purchase because you already own it. You will not even get to the screen to choose a payment method. What will happen is that the store will notify the app that it is already purchased, which will unlock the app and activate it.

This workaround is not nice, and I understand the concern here (the user may still be afraid to try purchasing a second time), and it seems to be reported more than once... 

So I decided to work on a solution for it now as a hotfix, and I will release a new version with a hotfix in a few days.

**Comment by gileli121** [comment]
And I have good news - I could reproduce the bug on my end.
You need to start the app without an internet connection to reproduce it (and use the store activation method, not the serial key)

This is a regression bug that started to happen after the key-activation method was added to the MS store version.

The hotfix is very simple, so that I will release a new version with the fix very soon (probably today or tomorrow)

**Comment by gileli121** [comment]
@benrobbins20 
The issue has been fixed for the store version.
It is fixed in v5.19.5, and this version is released only on Microsoft Store because all the fixes are only relevant to the store version.

The update will be available during the next day or in a few hours. It is waiting for Microsoft approval.

Update notes:

```
This release version is only available in the Microsoft Store because all bugfixes here are related to the activation/license system that, for some customer ware function incorrectly.

Bugfixes:

* Fixed a bug that if you activate it by the store, the app may deactivate at some point, and it can happen if, for example, you don't have an internet connection. The bug caused the app to ignore its activation information (that is stored in the cache) and, as a result, recheck it with the MS store API when the check failed due to an internet connection (or maybe a different issue). This triggers deactivation.
* Customers who reinstalled windows were unable to figure out how to reuse the activation by the store. We fixed it by renaming the button "Purchase (Microsoft Store)" to "Purchase or Login (Microsoft Store)," and we also added a tooltip for this button with an explanation that clarifies that users can reuse the license by clicking on this button and logging in.
* Small minor change: once the user logged in to activate the app, there was an error notification that the app had already been purchased and was now activated. The error notification changed to an information notification (because this is not a real error)
```
